code/modules/surgery/organs/autoimplanter.dm

```diff
@@ -7,18 +7,27 @@
 	w_class = WEIGHT_CLASS_SMALL
 	usesound = 'sound/weapons/circsawhit.ogg'
 	var/uses = -1  // -1 is infinite as far as we care
+	var/implanttime = 5 //fast for syndicate, .5sec
 	var/obj/item/organ/internal/cyberimp/storedorgan
+	var/possible_implants = list() //can put implant type references in here to let the user pick which one they want
 
 /obj/item/autoimplanter/attack_self(mob/user)//when the object it used...
 	if(uses == 0) // uses check before stored check here because a single use implanter probably wont have an implant in it after its been used up.
 		to_chat(user, "<span class='notice'>[src] appears to be permanently jammed.</span>")
 		return
-	if(!storedorgan)
+	if(!storedorgan && length(possible_implants) > 0)
+		var/selectedimplant = input("Select what to program the implant as.","Program Implant",/obj/item/organ/internal/cyberimp/eyes/hud/medical) in possible_implants
```

Comment:
This results in showing the user type paths, and I'm not quite sure how I fix it.